fetcher.go

```diff
@@ -6,31 +6,34 @@ import (
 	"fmt"
 	"io"
 
-	"github.com/ipld/go-ipld-prime/traversal/selector/builder"
-
-	"github.com/ipld/go-ipld-prime/traversal"
-
-	"github.com/ipld/go-ipld-prime/traversal/selector"
-
 	"github.com/ipfs/go-bitswap"
-
 	"github.com/ipfs/go-cid"
 	"github.com/ipld/go-ipld-prime"
 	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
 	basicnode "github.com/ipld/go-ipld-prime/node/basic"
+	"github.com/ipld/go-ipld-prime/traversal"
+	"github.com/ipld/go-ipld-prime/traversal/selector"
+	"github.com/ipld/go-ipld-prime/traversal/selector/builder"
 )
 
 // TODO: need to support sessions
 
 type Fetcher struct {
-	Exchange *bitswap.Bitswap
+	exchange *bitswap.Bitswap
```

Comment:
hey I just realized -- this really should not be an Exchange. it should be a block service, no? Unless we want to have more specific control over local vs remote loading I think?

Reply:
Actually I believe specifically this should be a blockservice.BlockGetter -- which covers both the session AND non session version

Reply:
This will be addressed in #2 .